ext/json/ext/generator/generator.c

```diff
@@ -1084,6 +1108,10 @@ static void generate_json(FBuffer *buffer, struct generate_json_data *data, JSON
                 if (klass != rb_cString) goto general;
                 generate_json_string(buffer, data, state, obj);
                 break;
+            case T_SYMBOL:
+                if (klass != rb_cSymbol) goto general;
+                generate_json_string(buffer, data, state, rb_sym2str(obj));
```

Comment:
Shouldn't it be `generate_json_symbol()` too?
Or at least I think static & dynamic symbols should be treated consistently.

Reply:
Yes it should. It's a small mistake. I need to go over the JRuby and TruffleRuby implementations anyway.